### ISSUE TITLE
Context menu to send links to kodi

### ISSUE THREAD

**Issue by arigit** [willhelm-mueller/playitonkodi#1]
Hi Willhelm, wondering if it would possible to support the following use caes in this addon:

Within firefox, right click on any link; select "Send to Kodi", then send the link to the playit listener as usual. Right now, your addon only supports clicking on the button and sending the current page URL which doesn't fit this use case

I use playit to play & queue audio podcasts and it would be great to be able to queue podcasts in my kodi instance by right clicking on firefox in my desktop. 

Thanks!

**Comment by willhelm-mueller** [comment]
Hi arigit,

*rightklick
this is a good Idea.I need to lookup how to add menuitems to the context menu.  I guess google will help me here.

*enquue files to playlist:
I already had something similar in mind: 
http://forum.kodi.tv/showthread.php?tid=118251&pid=2128295#pid2128295
I do not know how enqueing works on the Playit side. AFAIK there was a prior FF addon which was broken after FF 38 due to compatibilty issues. It could be a simple json parameter which differs between direct play and queueing media on KODI.

The lack of a working FF addon was the reason y i have written this addon.  

I also need to remove android and from the compatibilty list ;)

###
Update i found on my KODI 
.kodi/addons/service.playitx-0.6.3/config/turtle.xml
playHostedVideo 

maybe here i can start

**Comment by arigit** [comment]
I sketched something and it kind of works, the below could help.
Note that I bolted this into your addon and did some refactoring but it is quick & dirty.

Note also that with this, the button sends the Page URL to kodi, while the right-click option only shows when right-clicking on LINKS and then sends the link URL directly

```
var contextMenu = require("sdk/context-menu");
var menuItem = contextMenu.Item({
  label: "Send to Kodi",
  context: contextMenu.SelectorContext("a[href]"),
  contentScript: 'self.on("click", function (node, data) {' +
                 '  console.log("Item clicked!: " + node.href);' +
                 '  self.postMessage(node.href);' +
                 '});',
  onMessage: function (myurl) {
    mySendToKodi(myurl);
  }              
});

function mySendToKodi(myurl) {
    url =  myurl
    kodi_url = 'http://'+preferences['kodi_ip']+':'+preferences['kodi_port']+'/PlayIt';
    data =  {
                    "version":"1.1",
                    "method": "playHostedVideo",
                    "id"    : "1",
                    "params": {"videoLink" : url}
            }
    dataJSON = JSON.stringify(data);
    var post_rq = Request({
      url: kodi_url,
      content: dataJSON,
      onComplete: function (response) {
        notifications.notify({
            text: 'Sending: '+url+' to: '+kodi_url,
            iconURL : "resource://@sendtokodi/icons/k32.png"
        });
      }
    }).post();
};
```

**Comment by willhelm-mueller** [comment]
Thank you, looks great thanks! 
I will have a look on this.

**Comment by willhelm-mueller** [comment]
Context menu is now available in version 1.1.1
